cluster/sync-olm.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash -x
+
+# sync-olm syncs local code changes to the k8s/OCP cluster with OLM enabled on it.
+# It aims to be the single command that a dev needs to run to sync changes
+# instead of manually running a combination of below commands:
+# - make build-operator
+# - make build-webhook
+# - make build-manifests
+# - make container-build
+# - make container-push
+# - Replacing image names in couple of files
+# - make bundleRegistry
+# - ./hack/build-index-image.sh
+#
+# Prerequisites:
+# - Following variables to build container images and push them to a container registry.
+#   - QUAY_USERNAME: username to be used to login to the quay registry
+#   - QUAY_PASSWORD: password to authenticaye QUAY_USERNAME with quay registry. Helpful documentation - https://docs.quay.io/glossary/robot-accounts.html
+#   - IMAGE_TAG: The tag to be used for index image
+#   - REGISTRY_NAMESPACE: Your namespace on quay.io
+#   - IMAGE_REGISTRY: quay.io or the image registry you're using
+#   - CONTAINER_TAG: The tag to be used for hyperconverged-cluster-operator image
+# - Access to a k8s/OCP cluster that has OLM enabled on it.
+
+# tag for the bundle image
+CONTAINER_TAG="${CONTAINER_TAG:-$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | fold -w 8 | head -n 1)}"
+IMAGE_REGISTRY="${IMAGE_REGISTRY:-quay.io}"
+OPERATOR_GROUP_NAME=kubevirt-hyperconverged-group
+VERSION=1.10.0
+
+if [ -z "${REGISTRY_NAMESPACE}" ]; then
+  echo "Please set REGISTRY_NAMESPACE"
+  echo "   REGISTRY_NAMESPACE=kubevirt make cluster-sync-olm"
+  exit 1
+fi
+
+make build-operator
+make build-webhook
+make container-build
+make container-push
+
+if [ -z "${CSV_VERSION}"]; then
+  CSV_VERSION=latest
+fi
+# Image to be used in CSV manifests
+HCO_OPERATOR_IMAGE=$IMAGE_REGISTRY/$REGISTRY_NAMESPACE/hyperconverged-cluster-operator:$CONTAINER_TAG
+HCO_OPERATOR_IMAGE=$HCO_OPERATOR_IMAGE CSV_VERSION=$CSV_VERSION make build-manifests
+make bundleRegistry
+./hack/build-index-image.sh IMAGE_TAG
```

Comment:
$CONTAINER_TAG ?

Reply:
in this going to work if $IMAGE_REGISTRY/$REGISTRY_NAMESPACE/hyperconverged-cluster-index does not already exist?

Reply:
> `$CONTAINER_TAG` ?

I'll admit that I'm confused where exactly `CONTAINER_TAG` makes sense and where `IMAGE_TAG` makes sense. At this particular point, `IMAGE_TAG` seemed more appropriate to me.

> in this going to work if $IMAGE_REGISTRY/$REGISTRY_NAMESPACE/hyperconverged-cluster-index does not already exist?

I don't understand your question. If you are referring to the variables `$IMAGE_REGISTRY`, `$REGISTRY_NAMESPACE` and `$IMAGE_TAG` then I believe  `./hack/build-index-image.sh IMAGE_TAG` particularly depends on `$IMAGE_TAG` and won't work if it's not provided:
```sh
$ /hack/build-index-image.sh
+++ dirname ./hack/build-index-image.sh
++ readlink -e ./hack/../
+ PROJECT_ROOT=/home/dshah/src/hyperconverged-cluster-operator
+ DEPLOY_DIR=/home/dshah/src/hyperconverged-cluster-operator/deploy/olm-catalog
+ OUT_DIR=/home/dshah/src/hyperconverged-cluster-operator/_out
+ PACKAGE_NAME=community-kubevirt-hyperconverged
+ IMAGE_REGISTRY=quay.io
+ REGISTRY_NAMESPACE=dharmit
+ BUNDLE_REGISTRY_IMAGE_NAME=hyperconverged-cluster-bundle
+ INDEX_REGISTRY_IMAGE_NAME=hyperconverged-cluster-index
+ OPM=opm
+ UNSTABLE=
+ [[ 0 -gt 2 ]]
+ [[ 0 -lt 1 ]]
+ help
+ echo 'usage ./hack/build-index-image.sh {ALL|LATEST|<version>} [UNSTABLE]'
usage ./hack/build-index-image.sh {ALL|LATEST|<version>} [UNSTABLE]
+ exit 1
```

Reply:
1. `IMAGE_TAG` is a string where the value is `IMAGE_TAG`, if you want to reference a variable named `IMAGE_TAG` you have to use `${IMAGE_TAG}` and the shell parameter expansion will expand it.

2. isn't it going to try to pull `$IMAGE_REGISTRY/$REGISTRY_NAMESPACE/hyperconverged-cluster-index` first?

Reply:
> can we automate it somehow?

Yeah, I think so. I'll add it and ping you back.